### PR TITLE
new: Address VPC API changes

### PR DESCRIPTION
**Pull request by lgarber-akamai**
## 📝 Description

This PR implements various internal VPC API changes in the following data sources and resources:

- `linode_instance_config`
  - Expose `active` on network interfaces
- `linode_instance_ip`, `linode_instance_networking`
  - Expose `vpc_nat_1_1` if a VPC is NAT 1:1 mapped to an IP
- `linode_vpc_subnet`, `linode_vpc_subnets` 
  - Update the `linodes` field to contain a list of objects rather than a list of integers.
 
This PR makes some large alterations to the VPC Subnet acceptance test suite, specifically moving the VPC setup during init to the corresponding test templates. 

Depends on https://github.com/linode/linodego/pull/415

## ✔️ How to Test

Automated Testing:

```
make PKG_NAME=linode/instanceconfig testacc
make PKG_NAME=linode/instanceip testacc
make PKG_NAME=linode/vpcsubnet testacc
make PKG_NAME=linode/vpcsubnets testacc
```

Manual Testing:

**TODO**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -97,6 +97,6 @@ require (
 )
 
 // TODO: Remove this replacement before merging to dev
-replace github.com/linode/linodego => github.com/linode/linodego v1.24.1-0.20231024013514-a3629785d59b
```
Comment (lgarber-akamai) [inline go.mod:100]: We should revert this before merging.

File: linode/vpcsubnet/framework_models.go [hunk 2]
```diff
@@ -19,17 +20,63 @@ type VPCSubnetModel struct {
...
+			return nil, d
+		}
+
+		ifaces[i] = ifaceObj
+	}
+
+	ifacesList, d := types.ListValueFrom(ctx, LinodeInterfaceObjectType, ifaces)
+	if d.HasError() {
+		return nil, d
+	}
```
Comment (lgarber-akamai) [inline linode/vpcsubnet/framework_models.go:50]: I wonder if there's a cleaner way to create Framework objects and lists 🤔

File: linode/instanceconfig/resource_test.go [hunk 3]
```diff
@@ -350,22 +351,29 @@ func TestAccResourceInstanceConfig_vpcInterface(t *testing.T) {
...
+					resource.TestCheckResourceAttrSet(resName, "interface.1.ipv4.0.nat_1_1"),
+
+					resource.TestCheckResourceAttr(networkDSName, "ipv4.0.public.0.vpc_nat_1_1.address", "10.0.4.250"),
+					resource.TestCheckResourceAttrSet(networkDSName, "ipv4.0.public.0.vpc_nat_1_1.vpc_id"),
+					resource.TestCheckResourceAttrSet(networkDSName, "ipv4.0.public.0.vpc_nat_1_1.subnet_id"),
 				),
 			},
 			{
-				Config: tmpl.VPCInterfaceUpdates(t, instanceName, testRegion),
+				Config: tmpl.VPCInterfaceUpdates(t, instanceName, "us-east"),
```
Comment (ykim-1) [inline linode/instanceconfig/resource_test.go:368]: Should this be reverted back to `testRegion`?

File: linode/instanceconfig/resource_test.go [hunk 4]
```diff
@@ -350,22 +351,29 @@ func TestAccResourceInstanceConfig_vpcInterface(t *testing.T) {
 		CheckDestroy:             checkDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: tmpl.VPCInterface(t, instanceName, testRegion),
+				Config: tmpl.VPCInterface(t, instanceName, "us-east"),
```
Comment (ykim-1) [inline linode/instanceconfig/resource_test.go:354]: Same comment as in line 368

Comment (jriddle-linode) [inline linode/instanceconfig/resource_test.go:354]: Once VPC is out i think we can move this back to `testRegion`.

**Review by jriddle-linode — APPROVED** [review]

**Review by ykim-1 — APPROVED** [review]
Looks good! tested locally on both prod and beta

**Review by lgarber-akamai — COMMENTED** [review]
@zliang-akamai Thanks for refining this PR, all changes seem to work as expected. I just confirmed that all tests pass as expected against prod and the latest commit in `linodego proj/vpc`.

**Pull request merged**